extended/src/main/java/apoc/custom/CypherProceduresHandler.java

```diff
@@ -365,8 +367,39 @@ public RawIterator<AnyValue[], ProcedureException> apply(org.neo4j.kernel.api.pr
         }
     }
 
+    private synchronized void unregisterAndRegisterHomonymOne(Object signature, boolean override) {
+        boolean isProcedure = signature instanceof ProcedureSignature;
+        QualifiedName signatureName = isProcedure ? ((ProcedureSignature) signature).name() : ((UserFunctionSignature) signature).name();
+        // we unregister the previous function/procedure, if any
+        globalProceduresRegistry.unregister(signatureName);
+
+        // the 1st time (i.e. with override = true) we search for an analogous function/procedure as below
+        if (!override) {
+            return;
+        }
+
+        // if in sys-db there is a Procedure with the same name of the Function, or vice versa,
+        // we re-register it, because the above unregister method delete every fun and proc with that name.
+        // we use `registerWithoutOverride` to avoid recursion
+        readSignatures().filter(i -> {
```

Comment:
Who guarantees that here there are both the procs/funcs?
I mean you're removing at line 374 the proc/func with signature `custom.map_data`, who guarantees that the same procedure is not returned by `readSignatures`

Reply:
We need to exclude the `signatureName` from the records returned by `readSignatures`